File: ModernJavaInAction-master/src/main/java/modernjavainaction/chap20/Student.java
package modernjavainaction.chap20;

public class Student {

  private String name;
  private int id;

  public static void main(String[] args) {
    Student s = new Student("Raoul", 1);
    System.out.println(s.name);
    s.id = 1337;
    System.out.println(s.id);
  }

  public Student() {}

  public Student(String name, int id) {
    this.name = name;
    this.id = id;
  }

  public Student(String name) {
    this.name = name;
  }

  public String getName() {
    return name;
  }

  public void setName(String name) {
    this.name = name;
  }

  public int getId() {
    return id;
  }

  public void setId(int id) {
    this.id = id;
  }

}

File: ModernJavaInAction-scala/src/modernjavainaction/chap20/Student.scala
package modernjavainaction.chap20

class Student(var name: String, var id: Int)

object Student {

  def main(args: Array[String]) {
    val s = new Student("Raoul", 1)
    println(s.name)
    s.id = 1337
    println(s.id)

  }

}
